substrate/frame/ranked-collective/src/lib.rs

```diff
@@ -1003,5 +1003,12 @@ pub mod pallet {
 		fn demote(who: &Self::AccountId) -> DispatchResult {
 			Self::do_demote_member(who.clone(), None)
 		}
+
+		fn max_rank() -> Self::Rank {
+			MemberCount::<T, I>::iter()
+				.map(|(rank, _)| rank)
+				.max()
+				.unwrap_or(Self::min_rank())
```

Comment:
This is a very inefficient implementation. Generally I would expect that the `Config` trait has some `MaxRank` that could be returned here.